### ISSUE TITLE
TRUNCATE TABLE XXX is not working on v2.2

### ISSUE THREAD

**Issue by eranhazout** [scylladb/scylla#3694]
1. Create a simple table:
`CREATE TABLE gurushots.test5 (
   member_id ascii,
   challenge_id int,
   views counter,
   votes counter,
   PRIMARY KEY (member_id, challenge_id)
) WITH CLUSTERING ORDER BY (challenge_id DESC);`
2. Perform some random inserts (updates for counters in my case) of 200 rows
3. Using truncate table command doesn’t work, the data is still there

Additional info:
I run the "truncate table" command by cqlsh
When I run it from client it works after 4-6 times and the table was truncated
Using simple strategy and RF=3
experimental mode is enabled

Scylla version: upgraded cluster from 2.1 to 2.2
Cluster size: 3 nodes
OS: CentOS

*Hardware details (for performance issues)*          Delete if unneeded
Platform (physical/VM/cloud instance type/docker):
Hardware: cores=16 each memory=120GB each
Disks: 1 SSD each

**Comment by slivne** [comment]
@eranhazout - sorry for the late response.

I tried to reproduce locally on a 3 node cluster and failed - can you please share the journalctl logs.

**Comment by eranhazout** [comment]
Sorry, already killed this cluster.

**Comment by dorlaor** [comment]
@slivne another user complained about it, worth to check: https://stackoverflow.com/questions/52199363/truncate-the-table-in-scylla-loading-back

**Comment by skpotnuru** [comment]
Thank you @dorlaor for referencing me in this page. I am the one complained about the issue in stackoverflow.
I have a three node cluster. and I truncated a table using command

    truncate table students;

the command executing fine, means no error I am getting. If I query the table, the data is loading back. I think this is the ghost replication happening in my case. I even tried using all levels of consistency. But data still remains after performing truncation. I found a solution that to perform compaction operation after truncation, I even performed using nodetool in all the three nodes. But data still remains.
And I also observed that sometimes truncate working fine. In a single shot I can able to truncate the table. But sometimes even though I execute truncate table command multiple times data remains.
Some one please help me in this issue.
Thank you.

**Comment by skpotnuru** [comment]
Today I found some thing that, I truncate table and compact it using nodetool on all the three nodes. And two nodes data was removed and in one node data remains. After some time data is replicated back on two other nodes also which were empty after after compaction.

**Comment by slivne** [comment]
@skpotnuru can you please upload log files from the 3 nodes - if the logs are large you can follow https://docs.scylladb.com/operating-scylla/troubleshooting/report_scylla_problem/

**Comment by tomazbracic** [comment]
I have the same issue on production 2.2.0 system. Truncate work.

**Comment by tgrabiec** [comment]
I analyzed data provided by @tomazbracic, and the problem seems to be that one of the replicas does not process the truncate request, so it still has the data. The request doesn't seem to be received, it's not the case that it's blocked somewhere.

By analyzing the core dumps I've noticed that none of the nodes sent any requests to themselves, which implies that the truncate coordinator didn't consider itself as an eligible target for truncate. `gossiper::get_live_members()` will exclude itself when in shutdown state. And in fact, some shards of one of the nodes have STATUS set to `shutdown` for the current node on non-zero shards:

```
shard 0:
  gms::application_state::STATUS: {version=19, value="NORMAL,9133786472009474734"}
shard 1:
  gms::application_state::STATUS: {version=68230, value="shutdown,true"}
shard 2:
  gms::application_state::STATUS: {version=68230, value="shutdown,true"}
shard 3:
  gms::application_state::STATUS: {version=68230, value="shutdown,true"}
```

The local node is not actually shutting down.

After restarting the node, truncate succeeds.

I am figuring out how could we end up in such a state.
